File: apps/desktop/src-tauri/src/models/session.rs
use serde::{Deserialize, Serialize};

use super::agent::AgentMessage;

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct SessionHeader {
    pub id: String,
    pub title: String,
    pub cwd: String,
    pub project_id: Option<String>,
    pub model_id: String,
    pub provider: String,
    pub created_at: f64,
    pub updated_at: f64,
    pub message_count: Option<u64>,
    pub status: Option<String>,
    pub approval_mode: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct SessionDetailResponse {
    pub header: SessionHeader,
    pub messages: Vec<AgentMessage>,
}

File: apps/console-macos/Sources/ConsoleCore/Models/Session.swift
import Foundation

// MARK: - SessionHeader  (mirrors models/session.rs)

public struct SessionHeader: Codable, Sendable, Hashable, Identifiable {
    public let id: String
    public var title: String
    public let cwd: String
    public let projectId: String?
    public let modelId: String
    public let provider: String
    public let createdAt: UInt64
    public let updatedAt: UInt64
    public let messageCount: UInt64?

    enum CodingKeys: String, CodingKey {
        case id
        case title
        case cwd
        case projectId   = "projectId"
        case modelId     = "modelId"
        case provider
        case createdAt   = "createdAt"
        case updatedAt   = "updatedAt"
        case messageCount = "messageCount"
    }
}

// MARK: - SessionDetailResponse

public struct SessionDetailResponse: Codable, Sendable, Hashable {
    public let header: SessionHeader
    public let messages: [AgentMessage]
}
